endoreg_db/views/pdf/pdf_stream.py

```
@@ -0,0 +1,186 @@
+                        logger.error(
+                            f"Error opening {file_type} PDF file for range request: {e}"
+                        )
+                        raise Http404(f"Error accessing {file_type} PDF file")
+                else:
+                    logger.warning(f"Invalid Range header format: {range_header}")
+
+            # Serve entire file using FileResponse (automatically handles file closing)
+            logger.debug(f"Serving full {file_type} PDF {pk} ({file_size} bytes)")
+
+            try:
+                file_handle = open(file_path, "rb")
```

Pylint (4.0.2)

[refactor] Consider using 'with' for resource-allocating operations

(R1732)